system-addon/test/functional/mochitest/browser_topsites_section.js

```diff
@@ -54,3 +54,52 @@ test_newtab({
       "Topsite should be unpinned");
   }
 });
+
+// Check Topsites add
+test_newtab(
+  // it should be able to click the topsites edit button to reveal the edit topsites modal and overlay.
+  async function topsites_edit() {
+    // Find the edit topsites button
+    const topsitesEditBtn = content.document.querySelector(".edit-topsites-button button");
+    topsitesEditBtn.click();
+
+    let found = content.document.querySelector(".edit-topsites");
+    ok(found && !found.hidden, "Should find a visible topsites edit menu");
+
+    found = content.document.querySelector(".modal-overlay");
+    ok(found && !found.hidden, "Should find a visible overlay");
+
+    // Find the "Add" button
+    const topsitesAddBtn = content.document.querySelector(".add");
+    topsitesAddBtn.click();
+
+    // Write field title
+    let fieldTitle = content.document.querySelector(".field.title input");
+    ok(fieldTitle && !fieldTitle.hidden, "Should find field title input");
+
+    fieldTitle.value = "Youtube";
+    fieldTitle.dispatchEvent(new Event("input", {bubbles: true}));
+    is(fieldTitle.value, "Youtube", "The field title should match");
+
+    // Write field url
+    let fieldURL = content.document.querySelector(".field.url input");
+    ok(fieldURL && !fieldURL.hidden, "Should find field url input");
+
+    fieldURL.value = "https://www.youtube.com/";
+    fieldURL.dispatchEvent(new Event("input", {bubbles: true}));
+    is(fieldURL.value, "https://www.youtube.com/", "The field title should match");
+
+    // Click the "Add" button
+    let addBtn = content.document.querySelector(".done.add");
+    addBtn.click();
+
+    // Click the "Done" button
+    let doneBtn = content.document.querySelector(".done");
+    doneBtn.click();
+
+    // Wait for Topsite to be populated
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-site-outer:nth-child(1) a"), "No Topsite found");
+    let topSite = content.document.querySelector(".top-site-outer:nth-child(1) a");
```

Comment:
Could probably just do `:first-child` and have the condition to be waited for just be that the top site's href is the expected value.